examples/03_press_drag_release_tool/ex_press_drag_release_tool/main.rb

```diff
@@ -0,0 +1,182 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+
+module Examples
+  module PressDragReleaseTool
+
+    class LineTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      DRAG_THRESHOLD = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+
+        # Track where mouse was pressed down so we can compare its position when
+        # later released.
+        @mouse_down = ORIGIN
+
+        update_ui
+      end
+
+      def deactivate(view)
+        view.invalidate
+      end
+
+      def resume(view)
+        update_ui
+        view.invalidate
+      end
+
+      def suspend(view)
+        view.invalidate
+      end
+
+      def onCancel(reason, view)
+        reset_tool
+        view.invalidate
+      end
+
+      def onMouseMove(flags, x, y, view)
+        if picked_first_point?
+          @mouse_ip.pick(view, x, y, @picked_first_ip)
+
+          # Print the length of the previewed line to the measurement bar.
+          # Printing a Length object (as opposed to a float) automatically
+          # formats it according to the model units.
+          Sketchup.vcb_value = @mouse_ip.position.distance(@picked_first_ip.position)
```

Comment:
Did you see my previous not about the VCB value? The VCB will be cleared when the tool is suspended. So when you resume this will be lost. I suggested caching the VCB value and set it duing `update_ui`.